fakeredis/_fakesocket.py

```diff
@@ -1034,6 +1038,27 @@ def zscore(self, key, member):
         except KeyError:
             return None
 
+    @command(name="zmscore", fixed=(Key(ZSet), bytes), repeat=(bytes,))
+    def zmscore(self, key: str, *members: Union[str, bytes]) -> list[Optional[float]]:
+        """Get the scores associated with the specified members in the sorted set stored at key.
+        For every member that does not exist in the sorted set, a nil value is returned."""
+        members = (
```

Comment:
You can use `key.value[member]` to get member value when member is bytes (you don't need to convert members to str

Reply:
You can simplify the implementation like this:
```
res= [key.value[member] for member in members]
```

Reply:
Thinking about it, the implementation is effectively just delegating to `zscore`, so there's no need to pre-process `members` at all. I've removed that part of the implementation.

Reply:
Implementing a public method using another public method is not a good practice.

We can go this path by extracting `zscore` to `_zscore` and then implementing both `zscore` and `zmscore` using `_zcore`.

Reply:
Hadn't considered that. Would you prefer that? Otherwise I can simply change the implementation to use `FakeSocket._encodefloat` directly.

Reply:
I think using `FakeSocket._encodefloat` makes sense.
But you should also check for `None`s since `_encodefloat` doesn't deal with None.

Also write tests for Nones, etc.

Reply:
Implementation updated. The tests I wrote previously do account for missing members, so no worries there. I'll break them up so they don't rely on [`pytest-describe`](https://github.com/pytest-dev/pytest-describe).

Side note - I do feel pretty strongly that the test suite's overall readability and/maintainability would be dramatically improved by reorganizing it with `pytest-describe`. Using the extant `zscore` tests as an example, it would go from:

```python
def test_zscore(r):
    testtools.zadd(r, 'foo', {'one': 54})
    assert r.zscore('foo', 'one') == 54


def test_zscore_non_existent_member(r):
    assert r.zscore('foo', 'one') is None


def test_zscore_wrong_type(r):
    r.sadd('foo', 'bar')
    with pytest.raises(redis.ResponseError):
        r.zscore('foo', 'one')
```

to:

```python
@pytest.mark.description
def describe_zscore() -> None:
    """Test that `fakeredis`'s implementation of `zscore` behaves as expected
    when..."""

    key, member, score = "foo", "one", 54

    @pytest.mark.description
    def when_member_is_present(r: redis.Redis) -> None:
        """When the requested member is present in the sorted set, its score
        should be returned and the returned score should match the value that
        was originally set."""
        testtools.zadd(r, key, {member: score})
        assert r.zscore(key, member) == score

    @pytest.mark.description
    def when_member_is_not_present(r: redis.Redis) -> None:
        """When the requested member is not present in the sorted set, a score
        of `None` should be returned."""
        assert r.zscore(key, member) is None

    @pytest.mark.description
    def when_cached_value_is_not_a_sorted_set(r: redis.Redis) -> None:
        """When the cached value referred to by `key` is present but is not a
        sorted set, an error should be raised."""
        r.sadd(key, member)

        with pytest.raises(redis.ResponseError):
            r.zscore(key, member)
```

Granted, the second example *does* seem like _more_ but the symantics are much clearer and therefore easier to reason about. Happy to tackle this, if you're willing to sign off on it.

Reply:
I am totally onboard for using `pytest-describe`.
But please move hypothesis tests to `test_hypothesis.py` instead of `test_fakeredis6.py`

Reply:
For the `pytest-describe` - add it to the new tests only for this PR.

Reply:
I've already removed it in order to comply with your request to revert the `pyproject.toml`. For this PR, I think you're right - better to keep it simple. I'll add the `describe` tests to whatever working branch you assign me to once this initial "interview" PR is complete.

Does that seem fair? 😅

Reply:
You are not being interviewed :) 

I accept every PR I understand and align with how the package is built (See closed PRs, there are quite a few by other developers).
Small PRs make it easier for me to understand.